src/components/Button/Button.tsx

```diff
@@ -288,107 +365,31 @@ const Button = forwardRef(
-  /** Callback function to run when the `onMouseDown` event is triggered */
-  onMouseDown: PropTypes.func,
-  /** Blur the button after click */
-  blurOnMouseUp: PropTypes.bool,
-  /** The button's form name */
-  name: PropTypes.string,
-  /** The button's size  */
-  size: PropTypes.oneOf([Button.sizes.SMALL, Button.sizes.MEDIUM, Button.sizes.LARGE]),
-  /** The button's color  */
-  color: PropTypes.oneOf([
-    Button.colors.PRIMARY,
-    Button.colors.NEGATIVE,
-    Button.colors.POSITIVE,
-    Button.colors.ON_PRIMARY_COLOR,
-    Button.colors.ON_INVERTED_BACKGROUND
-  ]),
-  /** The button's type  */
-  type: PropTypes.oneOf([Button.inputTags.BUTTON, Button.inputTags.SUBMIT, Button.inputTags.RESET]),
-  /** Whether the button is disabled or not */
-  disabled: PropTypes.bool,
-  /** Icon to render to the right of the button */
-  rightIcon: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
-  /** Icon to render to the left of the button */
-  leftIcon: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
-  /** Displays a custom success state, see the `successIcon` and `successText` props as well */
-  success: PropTypes.bool,
-  /** Success icon name */
-  successIcon: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
-  /** Success text */
-  successText: PropTypes.string,
-  /** Displays a spinner on the button */
-  loading: PropTypes.bool,
-  /** Custom style to pass to the component */
-  style: PropTypes.object,
-  /** Marks the button as active */
-  active: PropTypes.bool,
-  /** HTML ID to pass to the button */
-  id: PropTypes.string,
-  /** Adds a little bit of margin to the right */
-  marginRight: PropTypes.bool,
-  /** Adds a little bit of margin to the left */
-  marginLeft: PropTypes.bool,
-  /** Element id to describe the button accordingly */
-  ariaLabeledBy: PropTypes.string,
-  /** Aria label to provide important when providing only Icon */
-  ariaLabel: PropTypes.string,
-  /** Aria for a button popup */
-  ariaHasPopup: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
-  /** Aria to be set if the popup is open */
-  ariaExpanded: PropTypes.bool,
-  /** Aria controls - receives id for the controlled region */
-  ariaControls: PropTypes.string,
-  /** Callback function to run when the button is focused */
-  onFocus: PropTypes.func,
-  /** Callback function to run when the button is blurred */
-  onBlur: PropTypes.func,
-  rightFlat: PropTypes.bool,
-  leftFlat: PropTypes.bool,
-  preventClickAnimation: PropTypes.bool,
-  noSidePadding: PropTypes.bool,
-  /** Default text color in `ON_PRIMARY_COLOR` kind (should be any type of css color (rgb, var, hex...) */
-  defaultTextColorOnPrimaryColor: PropTypes.string,
-  /** Change the focus indicator from around the button to within it */
-  insetFocus: PropTypes.bool
-};
+Object.assign(Button, {
+  sizes: SIZES,
+  colors: ButtonColor,
+  kinds: ButtonType,
+  types: ButtonInputType,
+  inputTags: ButtonInputType
+});
 
 Button.defaultProps = {
```

Comment:
Do we still need `defaultProps`? IMO it would be better to switch to native JS default values.

Reply:
I think we keep these because something which related to the storybook integration. maybe you remember ? @orrgottlieb @laviomri 
Anyway I will try to remove this and see what happend

Reply:
I agree with @sahariko, typescript will ignore those values anyway and it will be confusing if we'll continue using defaultProps

Reply:
We did not removing the default props for component which we using for fake inheritance (which is basically not inheritance but composition).
For example we have icon button which receive the same default props like the button and we want the storybook to be aware of it.
In regular component we will remove default props.

Reply:
@hadasfa I think the issue was that storybook didn't detect the TS types, and was looking for `defaultProps` to understand the types (and show them in the props table). Worth checking this out, since I agree with @eran-cohen and @sahariko

Reply:
I almost sure that in regular cases it will succeed to detect the default values, but either way it will not work for button composites (icon button etc) and dialog composites (tooltip etc...)
Will be checked as an option when I will convert a regular component